Utils/azuretools-core/src/com/microsoft/azuretools/authmanage/srvpri/SrvPriManager.java

```diff
@@ -207,46 +208,39 @@ public void write(String str) throws IOException {
         }
     }
 
-    private static void checkArtifact(Reporter<String> fileReporter, Path filePath) throws IOException {
+    private static Mono<Boolean> checkArtifact(Reporter<String> fileReporter, Path filePath) {
         // here we try to use the file to check it's ok with retry logic
         fileReporter.report("Checking cred file...");
-        final int RETRY_QNTY = 5;
-        final int SLEEP_SEC = 10;
-        int retry_count = 0;
-        File authFiel = new File(filePath.toString());
-        while (retry_count < RETRY_QNTY) {
+
+        int maxRetry = 4;
+        return Mono.fromCallable(() -> {
+            File authFile = new File(filePath.toString());
             try {
                 fileReporter.report("Checking: Azure.authenticate(authFile)...");
-                Azure.Authenticated azureAuthenticated = Azure.authenticate(authFiel);
+                Azure.Authenticated azureAuthenticated = Azure.authenticate(authFile);
                 fileReporter.report("Checking: azureAuthenticated.subscriptions().list()...");
                 azureAuthenticated.subscriptions().list();
                 fileReporter.report("Checking: azureAuthenticated.withDefaultSubscription()...");
                 Azure azure = azureAuthenticated.withDefaultSubscription();
                 fileReporter.report("Checking: resourceGroups().list()...");
                 azure.resourceGroups().list();
                 fileReporter.report("Done.");
-                break;
+                return true;
             } catch (Throwable e) {
                 LOGGER.info("=== checkArtifact@SrvPriManager exception: " + e.getMessage());
                 //e.printStackTrace();
                 if (needToRetry(e)) {
-                    retry_count++;
-                    if ((retry_count >= RETRY_QNTY)) {
-                        fileReporter.report(String.format("Failed to check cred file -retry limit %s has reached, error: %s", RETRY_QNTY, e.getMessage()));
-                        throw e;
-                    }
-                    LOGGER.info(String.format("Failed %d/%d, will retry in %s seconds, error: %s", retry_count, RETRY_QNTY, SLEEP_SEC, e.getMessage()));
-                    try {
-                        Thread.sleep(SLEEP_SEC * 1000);
-                    } catch (InterruptedException e1) {
-                        fileReporter.report("Interrupted sleep: " + e.getMessage());
-                    }
-                } else {
-                    fileReporter.report(String.format("Failed to check cred file after %s retries, error", retry_count, e.getMessage()));
+                    // make the retry work
                     throw e;
                 }
+                return false;
             }
-        }
+        }).retryWhen(Retry.backoff(maxRetry, Duration.ofSeconds(10))).onErrorResume(e -> {
+            fileReporter.report(String.format("Failed to check cred file after %s retries, error: %s", maxRetry, e.getMessage()));
```

Comment:
Do we need to distinguish whether we meet no need to retry exception or we have retried for max time but no result?